x/ecocredit/server/core/features/msg_mint_batch_credits.feature

```diff
@@ -1,67 +1,141 @@
 Feature: Msg/MintBatchCredits
 
   Credits can be dynamically minted to an existing credit batch:
+  - when the credit batch exists
   - when the credit batch is open
-  - when the origin tx id and source is unique within the scope of a credit class
-  - ...
+  - when the issuer is the issuer of the credit batch
+  - when the origin tx id and source is unique within the scope of the credit class
+  - when the contract is unique within the scope of the credit class
+  - the recipient batch balance is updated
+  - the batch supply is updated
+
+  Rule: The credit batch must exist
+
+    Background:
+      Given a credit type with abbreviation "C"
+      And a credit class with id "C01" and issuer alice
+      And a project with id "C01-001"
+      And a credit batch with denom "C01-001-20200101-20210101-001" and issuer alice
+
+    Scenario: the credit batch exists
+      When alice attempts to mint credits with batch denom "C01-001-20200101-20210101-001"
+      Then expect no error
+
+    Scenario: the credit batch does not exist
+      When alice attempts to mint credits with batch denom "C01-001-20200101-20210101-002"
+      Then expect the error "could not get batch with denom C01-001-20200101-20210101-002: not found: invalid request"
 
   Rule: The credit batch must be open
 
     Background:
-      Given a credit type
-      And a credit class with issuer alice
-      And a project
+      Given a credit type with abbreviation "C"
+      And a credit class with id "C01" and issuer alice
+      And a project with id "C01-001"
 
     Scenario: the credit batch is closed
-      Given a credit batch with open "false" and issuer alice
-      When alice attempts to mint credits
-      Then expect the error "unable to mint credits: credits cannot be minted in a closed batch: invalid request"
+      Given a credit batch with denom "C01-001-20200101-20210101-001" open "false" and issuer alice
+      When alice attempts to mint credits with batch denom "C01-001-20200101-20210101-001"
+      Then expect the error "credits cannot be minted in a closed batch: invalid request"
 
     Scenario: the credit batch is open
-      Given a credit batch with open "true" and issuer alice
-      When alice attempts to mint credits
+      Given a credit batch with denom "C01-001-20200101-20210101-001" open "true" and issuer alice
+      When alice attempts to mint credits with batch denom "C01-001-20200101-20210101-001"
       Then expect no error
 
+  Rule: The issuer must be the issuer of the credit batch
+
+    Background:
+      Given a credit type with abbreviation "C"
+      And a credit class with id "C01" and issuer alice
+      And a project with id "C01-001"
+      And a credit batch with denom "C01-001-20200101-20210101-001" and issuer alice
+
+    Scenario: the issuer is not the credit batch issuer
```

Comment:
same as the other review comment, not too sure on the wording here. it makes sense in the context of the message field, but if the reader doesn't have that context, could be confusing

Reply:
I agree. A bit awkward but leaving for now and tracking in the issue for writing guidelines: #1242